### PR TITLE
Ugc to master

### PR DESCRIPTION
**Pull request by VladiMihaylenko**

### CODE REVIEW

**Comment by VladiMihaylenko** [comment]
@alexzatsepin @ygorshenin @mgsergio ptal

File: base/worker_thread.cpp [hunk 1]
```diff
@@ -18,52 +18,56 @@ WorkerThread::~WorkerThread()
 
 void WorkerThread::ProcessTasks()
 {
-  queue<Task> pending;
+  unique_lock<mutex> lk(m_mu, defer_lock);
```
Comment (ygorshenin) [inline base/worker_thread.cpp]: No, I'd prefer to keep master version.

File: base/worker_thread.cpp [hunk 2]
```diff
@@ -18,52 +18,56 @@ WorkerThread::~WorkerThread()
...
 
   while (true)
   {
     Task task;
 
     {
-      unique_lock<mutex> lk(m_mu);
+      lk.lock();
```
Comment (ygorshenin) [inline base/worker_thread.cpp]: We should take the version of worker_thread.cpp from master.

File: base/worker_thread.hpp [hunk 3]
```diff
@@ -29,22 +28,18 @@ class WorkerThread
   WorkerThread();
   ~WorkerThread();
 
-  // Pushes task to the end of the thread's queue. Returns false when
-  // the thread is shut down.
   template <typename T>
-  bool Push(T && t)
+  void Push(T && t)
```
Comment (ygorshenin) [inline base/worker_thread.hpp]: You should take the version of worker_thread.hpp from master.

File: base/base_tests/worker_thread_tests.cpp [hunk 4]
```diff
@@ -19,12 +18,12 @@ UNIT_TEST(WorkerThread_Smoke)
...
   {
     WorkerThread thread;
-    TEST(thread.Shutdown(WorkerThread::Exit::SkipPending), ());
+    thread.Shutdown(WorkerThread::Exit::SkipPending);
   }
 
   {
     WorkerThread thread;
-    TEST(thread.Shutdown(WorkerThread::Exit::ExecPending), ());
+    thread.Shutdown(WorkerThread::Exit::ExecPending);
```
Comment (ygorshenin) [inline base/base_tests/worker_thread_tests.cpp]: The test must be taken from master.

**Comment by VladiMihaylenko** [comment]
@ygorshenin ptal

**Pull request merged**